sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+    gpu_instance_type: str
+    graph_construction_instance_type: str
+    gsprocessing_instance_count: int
+    train_on_cpu: bool
+    volume_size_gb: int
+
+    def __post_init__(self):
+        assert (
+            self.cpu_instance_type or self.gpu_instance_type
+        ), "At least one instance type (CPU or GPU) should be specified."
+
+        if self.train_on_cpu:
+            assert (
+                self.cpu_instance_type
+            ), "Need to provide a CPU instance type when training on CPU"
+
+        if not self.graph_construction_instance_type:
+            self.graph_construction_instance_type = self.cpu_instance_type
+            logging.warning(
+                "No graph processing instance type specified, using the CPU instance type: %s",
+                self.cpu_instance_type,
+            )
+
+
+@dataclass
+class TaskConfig:
+    """Pipeline/task-level configuration"""
+
+    base_job_name: str
+    graph_name: str
+    input_data_s3: str
+    jobs_to_run: List[str]
+    log_level: str
+    output_prefix: str
+    pipeline_name: str
+
+    def __post_init__(self):
+        # Ensure job names are valid
+        for job_type in self.jobs_to_run:
+            assert (
+                job_type in JOB_ORDER
+            ), f"Unsupported job type: '{job_type}', expected one of {list(JOB_ORDER.keys())}"
+
+        # Ensure jobs are in the right order
+        self.jobs_to_run.sort(key=lambda x: JOB_ORDER[x])
+
+        # We should only run either gconstruct or gsprocessing
+        if "gconstruct" in self.jobs_to_run and "gsprocessing" in self.jobs_to_run:
+            raise ValueError(
+                "Should not try to run both GConstruct and GSProcessing steps, "
+                f"got job sequence: {self.jobs_to_run}"
+            )
+        # TODO: Should we allow running GConstruct with dist_part?
+
+        # When running gsprocessing ensure we run dist_part as well
+        if "gsprocessing" in self.jobs_to_run and "dist_part" not in self.jobs_to_run:
+            raise ValueError(
+                "When running GSProcessing need to run 'dist_part' as the following job, "
+                f"got job sequence: {self.jobs_to_run}"
+            )
+
+        # Fix trailing slash for S3 input paths, otherwise can get
+        # ClientError: Failed to invoke sagemaker:CreateProcessingJob. Error Details: null.
+        if self.input_data_s3.endswith("/"):
+            self.input_data_s3 = self.input_data_s3[:-1]
+
+
+@dataclass
+class GraphConstructionConfig:
+    """Configuration for the graph construction step"""
+
+    config_filename: str
+    graph_construction_args: str
+
+
+@dataclass
+class PartitionConfig:
+    """Configuration for the partition step"""
+
+    partition_algorithm: str
+    input_json_filename: str
+    output_json_filename: str
+
+
+@dataclass
+class TrainingConfig:
+    """Configuration for the training step"""
+
+    model_output_path: str
+    train_inference_task: str
```

Comment:
Why we need train_inference_task here?

Reply:
It's the way we provide e.g. `node_classification` to the train/inference launch scripts, that requires the `--task-type` to be set.